src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -843,18 +855,22 @@ static int TopicPartition_init (PyObject *self, PyObject *args,
 	const char *topic;
 	int partition = RD_KAFKA_PARTITION_UA;
 	long long offset = RD_KAFKA_OFFSET_INVALID;
+	const char *metadata = NULL;
+
 	static char *kws[] = { "topic",
 			       "partition",
 			       "offset",
+			       "metadata",
 			       NULL };
-
-	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|iL", kws,
-					 &topic, &partition, &offset))
+        
+	if (PyArg_ParseTupleAndKeywords (args, kwargs, "s|iLy", kws,
```

Comment:
remove space before paren

Reply:
The `y` formatter token docs says:
> The bytes buffer must not contain embedded null bytes

But this is binary data, so it can contain anything. Need to use another formatter token here.